asset-api-management/apis/svc-asset-details/extension/clients/asset_details_client.go

```diff
@@ -0,0 +1,102 @@
+package clients
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"strings"
+	"svc-asset-details-layer/models"
+)
+
+type AssetDetailsClient struct {
+	rdsClient *RdsClient
+}
+
+const (
+	docIDKeyword            = "_docid.keyword"
+	docTypeKeyword          = "docType.keyword"
+	cloudType               = "_cloudType"
+	resourceId              = "_resourceid"
+	docId                   = "_docid"
+	underscoreDiscoveryDate = "_discoverydate"
+	discoveryDate           = "discoverydate"
+	firstDiscoveredOn       = "firstdiscoveredon"
+	entity                  = "_entity"
+	entityType              = "_entitytype"
+	loadDate                = "_loaddate"
+	assetRiskScore          = "assetRiskScore"
+)
+
+func NewAssetDetailsClient(configuration *Configuration) *AssetDetailsClient {
+	return &AssetDetailsClient{rdsClient: NewRdsClient(configuration)}
+}
+
+func (c *AssetDetailsClient) GetAssetDetails(ctx context.Context, ag string, targetType string, assetId string) (*models.AssetDetails, error) {
+	if len(strings.TrimSpace(assetId)) == 0 {
+		return nil, fmt.Errorf("assetId must be present")
+	}
+
+	result, err := FetchAssetDetails(ctx, ag, targetType, assetId, 1)
+
+	if err != nil {
+		return nil, err
+	}
+
+	sourceArr := (*result)["hits"].(map[string]interface{})["hits"].([]interface{})
+	if len(sourceArr) > 0 {
+		assetDetails := sourceArr[0].(map[string]interface{})
+
+		var assetDetailsFlat = make(map[string]interface{})
+		flatten("", &assetDetails, &assetDetailsFlat)
+
+		source := assetDetailsFlat[cloudType]
+		fieldsToBeSkipped := []string{resourceId, docId, underscoreDiscoveryDate, discoveryDate, firstDiscoveredOn, entity, entityType, loadDate, assetRiskScore}
```

Comment:
Let's define this as a constant list of fields and remove individual constants

Reply:
removed fieldsToBeSkipped as it is no longer need for asset model 2.0